Первая попытка

AntonKuzn opened the pull request.

AntonKuzn committed:
Первая попытка

nikolay-977 commented on .idea/encodings.xml:
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="Encoding">
+    <file url="file://$PROJECT_DIR$/src/main/java" charset="UTF-8" />
```
Необходимо исправить: в ПР-е не должно быть лишних файлов (.DS_Store, .idea, .build, .target за исключением файлов отчета). Это локальные файлы. Загружать локальные файлы на удаленный репозиторий не принято: это считается плохой практикой. Они не несут полезной информации, но занимают место.

nikolay977 replied:
не исправлено

nikolay-977 commented on src/test/java/GetListOrderTest.java:
```diff
@@ -0,0 +1,110 @@
+        response.then()
+                .body("success", equalTo(true))
+                .and()
+                .statusCode(200);
```
Необходимо исправить порядок проверки - сначала проверяем statusCode, затем остальное. Иначе в случае если сервер недоступен мы получим NullPointerException, а не status сode 500-й.

nikolay-977 commented on src/test/java/TestChangeUserInfo.java:
```diff
@@ -0,0 +1,81 @@
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.equalTo;
+
```
Можно улучшить: лишних пропусков строк следует избегать. Для зрительного разделения кода достаточно использовать один пропуск строки. Пропуски строк должны быть однообразными.

nikolay-977 reviewed (commented):
Можно улучшить: хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройка (если необходима) и как запускать (`mvn clean test` - кавычки позволяют выполнять запуск из файла README.md). Как оформлять README.md можно ознакомиться в статье https://blog.jetbrains.com/idea/2022/08/markdown-support-improvements/

AntonKuzn committed:
Поправил ошибки.
Исправил README, изменил порядок проверки статускод-остальные данны, поправил внешний вид кода (убрал лишние пропуски строк), удалил лишние файлы (которые смог).

AntonKuzn committed:
Delete .idea directory

AntonKuzn committed:
Delete target/test-classes directory

AntonKuzn committed:
Delete target/classes/org/example directory

AntonKuzn committed:
Delete target/maven-status/maven-compiler-plugin directory

AntonKuzn committed:
Delete .gitignore
